compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codemodifier/HttpPayloadParamIdentifier.java

```diff
@@ -0,0 +1,271 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.compiler.codemodifier;
+
+import io.ballerina.compiler.api.symbols.AnnotationSymbol;
+import io.ballerina.compiler.api.symbols.ArrayTypeSymbol;
+import io.ballerina.compiler.api.symbols.IntersectionTypeSymbol;
+import io.ballerina.compiler.api.symbols.MapTypeSymbol;
+import io.ballerina.compiler.api.symbols.ParameterSymbol;
+import io.ballerina.compiler.api.symbols.ResourceMethodSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TupleTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.projects.DocumentId;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.stdlib.http.compiler.HttpDiagnosticCodes;
+import io.ballerina.stdlib.http.compiler.HttpServiceValidator;
+import io.ballerina.stdlib.http.compiler.codemodifier.context.ParamAvailability;
+import io.ballerina.stdlib.http.compiler.codemodifier.context.ParamData;
+import io.ballerina.stdlib.http.compiler.codemodifier.context.PayloadParamContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.ballerina.stdlib.http.compiler.Constants.DEFAULT;
+import static io.ballerina.stdlib.http.compiler.Constants.GET;
+import static io.ballerina.stdlib.http.compiler.Constants.HEAD;
+import static io.ballerina.stdlib.http.compiler.Constants.OPTIONS;
+import static io.ballerina.stdlib.http.compiler.Constants.PAYLOAD_ANNOTATION_TYPE;
+import static io.ballerina.stdlib.http.compiler.Constants.QUERY_ANNOTATION_TYPE;
+import static io.ballerina.stdlib.http.compiler.HttpCompilerPluginUtil.updateDiagnostic;
+
+
+/**
+ * {@code HttpPayloadParamIdentifier} identifies the payload param during the initial syntax node analysis.
+ *
+ * @since 2201.5.0
+ */
+public class HttpPayloadParamIdentifier extends HttpServiceValidator {
+    private final Map<DocumentId, PayloadParamContext> payloadParamContextMap;
```

Comment:
Since this `payloadParamContextMap` is created for each service declarations and the documentId is same for one service. But we can have many resources which may need `payload` annotation.

So I think we should change this `Map<DocumentId, List<PayloadParamContext>>`. WDTY?

Reply:
Yes, I refactored the classes to be more descriptive and accurate. Please review again